app/controllers/uploads_controller.rb

```diff
@@ -0,0 +1,7 @@
+class UploadsController < AdminController
+  def get_presigned_url
```

Comment:
I have my thoughts on why you created the S3Connection class but do you mind explaining your thinking?

Reply:
I'm just making a service object that handles one thing: talking to AWS S3.

Maybe it's overkill here, but I'm trying to implement a best practice to avoid putting too much logic in controller classes. (You'd probably know more about this after reading POODR)